eclair-core/src/main/scala/fr/acinq/eclair/crypto/KeyManager.scala

```diff
@@ -73,5 +76,36 @@ trait KeyManager {
     */
   def sign(tx: TransactionWithInputInfo, publicKey: ExtendedPublicKey, remoteSecret: PrivateKey): ByteVector64
 
-  def signChannelAnnouncement(channelKeyPath: DeterministicWallet.KeyPath, chainHash: ByteVector32, shortChannelId: ShortChannelId, remoteNodeId: PublicKey, remoteFundingKey: PublicKey, features: ByteVector): (ByteVector64, ByteVector64)
+  /**
+    * Sign a channel announcement message
+    *
+    * @param fundingKeyPath BIP32 path of the funding public key
+    * @param chainHash chain hash
+    * @param shortChannelId short channel id
+    * @param remoteNodeId remote node id
+    * @param remoteFundingKey remote funding pubkey
+    * @param features channel features
+    * @return a (nodeSig, bitcoinSig) pair. nodeSig is the signature of the channel announcement with our node's
+    *         private key, bitcoinSig is the signature of the channel announcement with our funding private key
+    */
+  def signChannelAnnouncement(fundingKeyPath: DeterministicWallet.KeyPath, chainHash: ByteVector32, shortChannelId: ShortChannelId, remoteNodeId: PublicKey, remoteFundingKey: PublicKey, features: ByteVector): (ByteVector64, ByteVector64)
+
+  def signChannelAnnouncement(fundingKey: DeterministicWallet.ExtendedPublicKey, chainHash: ByteVector32, shortChannelId: ShortChannelId, remoteNodeId: PublicKey, remoteFundingKey: PublicKey, features: ByteVector): (ByteVector64, ByteVector64)
+  = signChannelAnnouncement(fundingKey.path, chainHash, shortChannelId, remoteNodeId, remoteFundingKey, features)
 }
+
+object KeyManager {
+  /**
+    * Create a BIP32 path from a public key. This path will be used to derive channel keys.
```

Comment:
It might be worth explaining in details why we derive channel keys from the funding key (what it brings for recovery when being funder, and that it doesn't help fundee that much).

Reply:
Yers, I'll update the comment, and we'll add (with another PR) a recovery tool that will allow users to get their funds back if all they have is their seed, the id of a node they had a channel to, and the channel id (which could be inferred from wallet transactions for "funder" channels).